lib/driver-parser.js

```diff
@@ -0,0 +1,119 @@
+const globalArgs = [
+  [['--json'], {
+    required: false,
+    defaultValue: false,
+    help: 'Use JSON for output format',
+    nargs: 0,
+    dest: 'json'
+  }]
+];
+
+const listArgs = [
+  ...globalArgs,
+  [['--installed'], {
+    required: false,
+    defaultValue: false,
+    help: 'List only installed drivers',
+    nargs: 0,
+    dest: 'installed'
+  }],
+  [['--updates'], {
+    required: false,
+    defaultValue: false,
+    help: 'Show information about newer versions',
+    nargs: 0,
+    dest: 'showUpdates'
+  }]
+];
+
+const installArgs = [
+  ...globalArgs,
+  [['driver'], {
+    type: 'string',
+    example: 'xcuitest',
+    help: 'Name of the driver to install',
+  }],
+  [['--npm'], {
+    required: false,
+    defaultValue: false,
+    nargs: 0,
+    help: 'Specify that the driver name provided is an NPM package',
+    dest: 'npm'
+  }],
+  [['--github'], {
+    required: false,
+    defaultValue: false,
+    nargs: 0,
+    help: 'Specify that the driver name provided is a GitHub repository of form org/repo',
+    dest: 'github'
+  }],
+  [['--git'], {
+    required: false,
+    defaultValue: false,
+    nargs: 0,
+    help: 'Specify that the driver name provided is a Git repository with the name being the Git repository to clone',
+    dest: 'git'
+  }]
+];
+
+const uninstallArgs = [
+  ...globalArgs,
+  [['driver'], {
+    type: 'string',
+    example: 'xcuitest',
+    help: 'Name of the driver to uninstall',
+  }],
+];
+
+const updateArgs = [
+  ...globalArgs,
+  [['driver'], {
+    type: 'string',
+    example: 'xcuitest',
+    help: 'Name of the driver to update, or the word "all" to update all. To see available updates, run "appium driver list --installed --updates"',
+  }],
+];
+
+function addDriverToParser (subParsers) {
+  const driverParser = subParsers.addParser('driver', {
+    addHelp: true,
+    help: 'Access the driver management CLI commands'
+  });
+  const driverSubParsers = driverParser.addSubparsers({dest: 'driverCommand'});
+
+  const listParser = driverSubParsers.addParser('list', {
+    addHelp: true,
+    help: 'List available and installed drivers'
+  });
+  for (const arg of listArgs) {
+    listParser.addArgument(arg[0], arg[1]);
+  }
+
+  const installParser = driverSubParsers.addParser('install', {
+    addHelp: true,
+    help: 'Install a driver'
+  });
+  for (const arg of installArgs) {
+    installParser.addArgument(arg[0], arg[1]);
+  }
+
+  const uninstallParser = driverSubParsers.addParser('uninstall', {
+    addHelp: true,
+    help: 'Uninstall a driver'
+  });
+  for (const arg of uninstallArgs) {
+    uninstallParser.addArgument(arg[0], arg[1]);
+  }
+
+  const updateParser = driverSubParsers.addParser('update', {
```

Comment:
Can we refactor this stuff? I observe a lot of duplication

Reply:
👍